src/test/ui/borrowck/borrowck-slice-pattern-element-loan-array.rs

```diff
@@ -0,0 +1,106 @@
+//compile-flags: -Z borrowck=mir
+
+#![feature(slice_patterns)]
+
+fn nop(_s: &[& i32]) {}
+fn nop_subslice(_s: &[i32]) {}
+
+fn const_index_ok(s: &mut [i32; 4]) {
```

Comment:
Would be good if we could split the test into one file for pass and one for fail.